keras_cv/layers/object_detection/retina_net_label_encoder.py

```diff
@@ -184,19 +190,12 @@ def call(self, images, boxes, classes):
             images=images[0],
         )
 
-        if isinstance(gt_boxes, tf.RaggedTensor):
-            gt_boxes = gt_boxes.to_tensor(default_value=-1, shape=(None, None, 4))
-        if isinstance(gt_classes, tf.RaggedTensor):
-            gt_classes = gt_classes.to_tensor(default_value=-1, shape=(None, None, 1))
-        elif gt_classes.get_shape().rank == 2:
-            gt_classes = gt_classes[..., tf.newaxis]
-
-        result = self._encode_sample(gt_boxes, gt_classes, anchor_boxes)
+        result = self._encode_sample(box_labels, anchor_boxes)
         result = bounding_box.convert_format(
             result, source="xywh", target=self.bounding_box_format, images=images
         )
-        encoded_box_targets = result[..., :4]
-        class_targets = result[..., 4]
+        encoded_box_targets = result["boxes"]
+        class_targets = result["classes"]
```

Comment:
It looks like now we're in a divergence -- previously every component accept tuples and returns tuples (same for model), but now some components accept dict input but return tuples.
Should we unify this? @ianstenbit

Reply:
lets solve in follow up